### ISSUE TITLE
`no-partial-of-curried` rule

### ISSUE THREAD

**Issue by jfmengels** [jfmengels/eslint-plugin-lodash-fp#22]
(Name to be improved)

Forbid the use `_.partial` when the method you're applying it to is curried.

Example
```js
const fn = _.partial(_.get, 'path'); // --> This is useless, could have done `_.get('path')`
const fn = _.partial(_.get, 'path', object);
// --> This will create a function that when called with of arguments, will simply return the function. Probably not the expected behavior.
```

**Comment by gunar** [comment]
Hey but then how would you do this?
```js
myRandomGenerator = _.partial(_.random, [1, 9])
```
Thanks!

**Comment by jfmengels** [comment]
I'd do the following
```js
myRandomGenerator = () => _.random(1, 9);
```

But you're on to something here... Seems like if you pass all the expected args, then it works. Pretty sure that previously, when you did that and then called it with no args, it'd return the function itself again (see the last example in the firrst code block here https://github.com/jfmengels/eslint-plugin-lodash-fp/blob/master/docs/rules/no-partial-of-curried.md).

@jdalton is that a behavior that has changed recently, or was I completely mistaken when I write this?
